itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestStatsReplicationScenarios.java

```diff
@@ -155,24 +182,50 @@ private void verifyNoStatsReplicationForMetadataOnly(String tableName) throws Th
     StatsSetupConst.setStatsStateForCreateTable(expectedFalseParams,
             replica.getTableColNames(replicatedDbName, tableName), StatsSetupConst.FALSE);
     Assert.assertTrue(rParams.equals(expectedFalseParams) || rParams.equals(expectedTrueParams));
+
+    verifyNoPartitionStatsReplicationForMetadataOnly(tableName);
+  }
+
+  private void verifyNoPartitionStatsReplicationForMetadataOnly(String tableName) throws Throwable {
+    // Test partition level stats
+    List<Partition> pParts = primary.getAllPartitions(primaryDbName, tableName);
+
+    if (pParts == null || pParts.isEmpty()) {
+      // Not a partitioned table, nothing to verify.
+      return;
+    }
+
+    // Partitions are not replicated in metadata only replication.
```

Comment:
In TableExport.getPartitions()
        if (tableSpec.specType == TableSpec.SpecType.TABLE_ONLY) {
          // TABLE-ONLY, fetch partitions if regular export, don't if metadata-only
          if (replicationSpec.isMetadataOnly()) {
            return null;

And dumpTable() always passes tableSpec with TABLE_ONLY set. So, we do not dump partitions in case of metadata only dump. So, we don't replication partition metadata as well in metadata-only dump.

Reply:
Yes... I know, currently we don't replicate partitions metadata if metadata-only. But, I think, this behaviour is incorrect and may have to discuss with team and conclude.